api.go

```diff
@@ -403,21 +403,17 @@ type CloudStorageClient interface {
 	SetBucketACL(bucketName string, cannedACL BucketACL) error
 	GetBucketACL(bucketName string) (BucketACL, error)
 
-	ListBuckets() ([]BucketStat, error)
-	ListObjects(bucket, prefix string, recursive bool, doneCh <-chan struct{}) <-chan ObjectStat
-	ListIncompleteUploads(bucket, prefix string, recursive bool, doneCh <-chan struct{}) <-chan ObjectMultipartStat
+	ListBuckets() ([]BucketInfo, error)
+	ListObjects(bucket, prefix string, recursive bool, doneCh <-chan struct{}) <-chan ObjectInfo
+	ListIncompleteUploads(bucket, prefix string, recursive bool, doneCh <-chan struct{}) <-chan ObjectMultipartInfo
 
 	// Object Read/Write/Stat operations.
-	GetObject(bucketName, objectName string) (reader io.ReadCloser, stat ObjectStat, err error)
-	PutObject(bucketName, objectName string, data io.Reader, size int64, contentType string) (n int64, err error)
-	StatObject(bucketName, objectName string) (ObjectStat, error)
+	GetObject(bucketName, objectName string) (reader *Object, err error)
+	PutObject(bucketName, objectName string, data io.Reader, contentType string) (n int64, err error)
```

Comment:
Rename "data -> reader". It will convey that ouput of GetObject can be passed here.